src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs

```diff
@@ -318,12 +318,46 @@ public IEnumerable<string> MakeCompliant(string name)
 
         private string CreateCompliantNameReusingPartialPrefixesAndSuffixes(string name)
         {
+            name = StripCommonPrefixes(name);
             name = EnsurePrefix(name);
             name = EnsureSuffix(name);
 
             return FinishFixingName(name);
         }
 
+        private static string StripCommonPrefixes(string name)
+        {
+            var index = 0;
+            while (name.Length > index + 1)
+            {
+                switch (char.ToLowerInvariant(name[index]))
+                {
+                case 'm':
+                case 's':
+                case 't':
```

Comment:
What about 'g'?

Reply:
For now I just used the prefixes deemed "common" by the old StyleCop implementation. We can add others if we want to, but probably as a separate issue. Want me to file a bug for this?